archr/targets/__init__.py

```diff
@@ -9,6 +9,11 @@
 import io
 import re
 
+try:
+    import snapshot
```

Comment:
What is this dependency? Can we get it marked as an extra_requires in setup.py?

Reply:
this is @Lukas-Dresel 's snapshot debugging code. I'll let him answer this question.

Reply:
This is for my snapshot debugging extension (which is really just glorified `fork()` with parents blocking to handle stdio). It's pushed here [Lukas-Dresel/snapshot](https://github.com/Lukas-Dresel/snapshot). It's super useful for debugging anything long running where it's easy to mess up the stepping through. 

One issue is that to revert a snapshot, it exits the child process and the parent process resumes as-is. However, exiting the child process causes the `archr` context managers to kill the `docker` containers it spun up, even though those are still used in all parent processes. This code only deletes the containers if we're not in a snapshot, so that only the final process exit will kill it. 

That being said, this is obviously far from perfect, and more one of a myriad of issues (sockets having had data sent, processes having been started/stopped, etc. Ideally for the ones we can (killing processes, containers) I'd like to have this here so that the only bugs that could be introduced are state-based ones, not random `docker container doesn't exist` which prevents me from debugging further.

Reply:
Seeing as it is debugging code, I am going to remove it before it goes to master.